src/sentry/sentry_apps/external_requests/select_requester.py

```diff
@@ -41,66 +49,83 @@ class SelectRequester:
-                    "select_options.requested",
-                    headers=self._build_headers(),
-                )
-            )
 
-            response = json.loads(body)
-        except Exception as e:
-            extra = {
+        with SentryAppInteractionEvent(
+            operation_type=SentryAppInteractionType.EXTERNAL_REQUEST,
+            event_type=SentryAppEventType.SELECT_OPTIONS_REQUESTED,
+        ).capture() as lifecycle:
+            extras: dict[str, Any] = {
                 "sentry_app_slug": self.sentry_app.slug,
                 "install_uuid": self.install.uuid,
                 "project_slug": self.project_slug,
             }
+            lifecycle.add_extras(extras)
+
+            try:
+                url = self._build_url()
+                extras.update({"url": url})
+
+                body = safe_urlread(
+                    send_and_save_sentry_app_request(
+                        url,
+                        self.sentry_app,
+                        self.install.organization_id,
+                        SentryAppEventType.SELECT_OPTIONS_REQUESTED,
+                        headers=self._build_headers(),
+                    )
+                )
+
+                response = json.loads(body)
+                extras.update({"response": response})
+            except RequestException as e:
+                halt_reason = f"{SentryAppEventType.SELECT_OPTIONS_REQUESTED}.{SentryAppExternalRequestHaltReason.BAD_RESPONSE}"
+                lifecycle.record_halt(halt_reason=e, extra={"reason": halt_reason, **extras})
 
-            if not url:
-                extra.update(
-                    {
-                        "uri": self.uri,
-                        "dependent_data": self.dependent_data,
-                        "webhook_url": self.sentry_app.webhook_url,
-                    }
+                raise SentryAppIntegratorError(
+                    message=f"Something went wrong while getting options for Select FormField from {self.sentry_app.slug}",
+                    webhook_context={"error_type": halt_reason, **extras},
+                    status_code=500,
+                ) from e
+
+            except Exception as e:
+                failure_reason = f"{SentryAppEventType.SELECT_OPTIONS_REQUESTED}.{SentryAppExternalRequestFailureReason.UNEXPECTED_ERROR}"
+                if not url:
+                    failure_reason = f"{SentryAppEventType.SELECT_OPTIONS_REQUESTED}.{SentryAppExternalRequestFailureReason.MISSING_URL}"
+                    extras.update(
+                        {
+                            "uri": self.uri,
+                            "dependent_data": self.dependent_data,
+                            "webhook_url": self.sentry_app.webhook_url,
+                        }
+                    )
+
+                raise SentryAppSentryError(
+                    message="Something went wrong while preparing to get Select FormField options",
```

Comment:
is the message used in the UI? should we still have `self.sentry_app.slug` in the message?

Reply:
It's not in the UI, Imo the slug being here wouldn't be very useful since this error would be on Sentry's side, so adding the 3p slug could confuse end users on who to ask for help.